packages/client/src/library/components/InputAddress.tsx

```diff
@@ -4,14 +4,31 @@ import { useAddressValidation } from 'hooks';
 import { formatAddress, isAddr } from 'utils';
 import Svg from 'library/Svg';
 
-const InputAddress = ({ web3, value, isValid, onChange }) => {
+export type AddressValidationResult = {
+  value: string;
+  isValid: boolean;
+};
+
+export interface InputAddressProps {
+  web3: any;
```

Comment:
Giving the web3 an any type since we are just starting the migration, there will be followup work to define the type for web3